src/main/js/distribution/Index.js

```diff
@@ -102,52 +99,37 @@ class Index extends Component {
 
     getCurrentJobConfig(currentRowSelected) {
         if (currentRowSelected != null) {
-            const { id, distributionType } = currentRowSelected;
-            if (distributionType === DescriptorUtilities.DESCRIPTOR_NAME.CHANNEL_EMAIL) {
-                return (<EmailJobConfiguration
-                    alertChannelName={distributionType}
-                    jobId={id}
+            const { id } = currentRowSelected;
+            return (
+                <DistributionConfiguration
                     handleCancel={this.cancelRowSelect}
-                    handleSaveBtnClick={this.saveBtn}
-                />);
-            } else if (distributionType === DescriptorUtilities.DESCRIPTOR_NAME.CHANNEL_HIPCHAT) {
-                return (<HipChatJobConfiguration
-                    alertChannelName={distributionType}
+                    onSave={this.saveBtn}
                     jobId={id}
-                    handleCancel={this.cancelRowSelect}
-                    handleSaveBtnClick={this.saveBtn}
+                    onModalClose={() => {
+                        this.props.fetchDistributionJobs();
+                        this.cancelRowSelect();
+                    }}
                 />);
-            } else if (distributionType === DescriptorUtilities.DESCRIPTOR_NAME.CHANNEL_SLACK) {
-                return (<SlackJobConfiguration
-                    alertChannelName={distributionType}
-                    jobId={id}
-                    handleCancel={this.cancelRowSelect}
-                    handleSaveBtnClick={this.saveBtn}
-                />);
-            }
         }
         return null;
     }
 
-    cancelRowSelect() {
-        this.startAutoReloadIfConfigured();
-        this.setState({
-            currentRowSelected: null
-        });
-    }
-
-    saveBtn() {
-        this.startAutoReloadIfConfigured();
-        this.cancelRowSelect();
-        this.reloadJobs();
-    }
-
-    reloadJobs() {
-        this.props.fetchDistributionJobs();
-    }
-
-    cancelAutoReload() {
-        clearTimeout(this.timeout);
+    createCustomModal(onModalClose, onSave, columns, validateState, ignoreEditable) {
+        return (
+            <DistributionConfiguration
+                projects={this.state.projects}
+                handleCancel={this.cancelRowSelect}
+                onModalClose={() => {
+                    this.props.fetchDistributionJobs();
+                    this.cancelRowSelect();
+                    onModalClose();
+                }}
+                onSave={this.saveBtn}
+                columns={columns}
```

Comment:
columns doesn't appear to be one of the properties on DistributionConfiguration. This should be removed until it is supported in DistributionConfiguration